### PR TITLE
574: git-info should use info from commit notifications

### PR DESCRIPTION
**Pull request by magicus**
git-info should use info from commit notifications to get the link to the review, not JBS (since not all PR have issues)

This required refactoring to move the hosted repo logic from cli/pr/Utils.java to where it can be accessed by all CLI tools. After some deliberating, I left a bunch of trivial wrappers in cli/pr/Utils.java -- otherwise this change would have touched *all* the files in cli/pr.

Testing: ad hoc testing of "git info" on some arbitrary Skara commits.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-574](https://bugs.openjdk.java.net/browse/SKARA-574): git-info should use info from commit notifications

### Reviewers
 * [Erik Helin](https://openjdk.java.net/census#ehelin) (@edvbld - **Reviewer**) ⚠️ Review applies to 50d827563e47142b011e9cee801e598112f5c2c3

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/1011/head:pull/1011`
`$ git checkout pull/1011`

### CODE REVIEW

File: forge/src/main/java/org/openjdk/skara/forge/HostedRepository.java [hunk 1]
```diff
@@ -82,4 +83,29 @@ default PullRequest createPullRequest(HostedRepository target,
...
+        }
+
+        /** The review comment looks like this:
+         * <!-- COMMIT COMMENT NOTIFICATION -->
+         * ### Review
+         *
+         * - [openjdk/skara/123](https://git.openjdk.java.net/skara/pull/123)
+         */
+
+        var pattern = Pattern.compile("### Review[^]]*]\\((.*)\\)");
```
Comment (edvbld) [inline forge/src/main/java/org/openjdk/skara/forge/HostedRepository.java:103]: Do you need to use `Pattern.DOTALL` here? As in `Pattern.compile("### Review[^]]*]\\((.*)\\)", Pattern.DOTALL);` ? I'm thinking about the `.*` handling `\n`.

Comment (magicus) [reply forge/src/main/java/org/openjdk/skara/forge/HostedRepository.java:103]: As you can see from the test, this works just fine. :-)

The `DOTALL` flag only applies to the dot `.`. I'm searching for "anything but ]", `[^]]` which also includes line terminators. And the URL (that I match with `.*`) should of course not have line breaks in it.

**Review by edvbld — APPROVED** [review]
Looks good modulo some whitespace issues found by jcheck :robot:

**Pull request closed without merging**